### PR TITLE
annotations: adding annotations for backendhttpsettings

### PR DESCRIPTION
**Pull request by akshaysngupta**
This PR adds annotations to modify http settings related to the backend connections.

| Annotation | Values | Default | BackendHttpSettings property |
| -- | -- | -- | -- |
| appgw.ingress.kubernetes.io/connection-draining | `bool` | `false` | ConnectionDraining.Enabled |
| appgw.ingress.kubernetes.io/connection-draining-timeout | `int32` | `30` | ConnectionDraining.DrainTimeoutInSec |
| appgw.ingress.kubernetes.io/cookie-based-affinity | `bool` | `false` | CookieBasedAffinity |
| appgw.ingress.kubernetes.io/request-timeout | `int32` | `30` | RequestTimeout |

Will add tests/documentation in following PRs.

### CODE REVIEW

File: pkg/appgw/backendhttpsettings.go [hunk 1]
```diff
@@ -18,6 +18,11 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const (
+	// DefaultConnDrainTimeoutInSec provides default value for ConnectionDrainTimeout
+	DefaultConnDrainTimeoutInSec = 30
```
Comment (draychev) [inline pkg/appgw/backendhttpsettings.go]: I wonder if we can get these consts from App Gateway on runtime so we don't replicate the values in this repo (just a thought)

File: pkg/appgw/backendhttpsettings.go [hunk 2]
```diff
@@ -182,5 +187,30 @@ func (builder *appGwConfigBuilder) generateHTTPSettings(backendID backendIdentif
 	if err == nil {
 		httpSettings.Path = to.StringPtr(pathPrefix)
 	}
+
+	isConnDrain, err := annotations.IsConnectionDraining(backendID.Ingress)
```
Comment (draychev) [inline pkg/appgw/backendhttpsettings.go]: You could (if you wanted to) inline this to a) limit the scope of the 2 variables and b) play some code golf:
```go
if isConnDrain, err := annotations.IsConnectionDraining(backendID.Ingress); if err == nil && isConnDrain {
```

File: pkg/appgw/backendhttpsettings.go (hunk 1 above)
Comment (akshaysngupta) [reply pkg/appgw/backendhttpsettings.go]: no we don't have any such api as of now :/

File: pkg/annotations/ingress_annotations_test.go [hunk 3]
```diff
@@ -79,7 +79,7 @@ func TestParseInt32Invalid(t *testing.T) {
 	value := "20asd"
 	ingress.Annotations[key] = value
 	parsedVal, err := parseInt32(&ingress, key)
-	if errors.IsInvalidContent(err) {
+	if !errors.IsInvalidContent(err) {
```
Comment (draychev) [inline pkg/annotations/ingress_annotations_test.go]: 🙀

Comment (akshaysngupta) [reply pkg/annotations/ingress_annotations_test.go]: missed it :/

**Comment by akshaysngupta** [comment]
Fixes #88

**Pull request merged**